ui/src/app/cohort-review/individual-participants-charts/individual-participants-charts.ts

```diff
@@ -0,0 +1,174 @@
+  @Input() chartData = [];
+  @Input() chartHeader: string;
+  trimmedData = [];
+  duplicateItems = [];
+  yAxisNames = [''];
+
+  constructor() {}
+
+  ngOnChanges() {
+    this.yAxisNames = [''];
+    if (this.chartData) {
+      this.setYaxisValue();
+    }
+
+  }
+
+  setYaxisValue() {
+    this.trimmedData = [];
+    this.chartOptions = {};
+    this.duplicateItems = [];
+    this.yAxisNames = [''];
+    let yAxisValue = 1;
+    this.chartData.reverse().map(items => { // find standardName in duplicate items
+     const duplicateFound = this.duplicateItems.find(
+       findName =>
+         findName.name === items.standardName
+       );
+      // duplicate items found return true otherwise push the the item in duplicateItems array
+      if (duplicateFound) {
+        Object.assign(items, {
+          yAxisValue: duplicateFound.yAxisValue,
+          startDate: moment(items.startDate, 'YYYY-MM-DD').unix() // format date to unix timestamp
+        });
+        return true;
+      }
+      this.duplicateItems.push({name: items.standardName, yAxisValue});
+      Object.assign(items, {
+        yAxisValue,
+        startDate: moment(items.startDate, 'YYYY-MM-DD').unix() // format date to unix timestamp
+      });
+      yAxisValue ++;
+    });
+    this.chartData.map(i => {
+      const temp = {
+        x: i.startDate,
+        y: i.yAxisValue,
+        standardName: i.standardName,
+        ageAtEvent: i.ageAtEvent,
+        rank: i.rank,
+        startDate: moment.unix(i.startDate).format('MM-DD-YYYY'),
+        standardVocabulary: i.standardVocabulary,
+      };
+      this.trimmedData.push(temp);
+      this.trimmedData.reverse();
+    });
+    this.duplicateItems.map(d => {
+      this.yAxisNames.push(d.name.substring(0, 13));
+    });
+
+    if (this.trimmedData.length) {
+      this.getChartsData();
+    }
+  }
+
+
+  getChartsData() {
+    const names = this.yAxisNames;
+    this.chartOptions = {
+           chart: {
+             type: 'scatter',
+             zoomType: 'xy',
+           },
+           credits: {
+             enabled: false
+           },
+           title: {
+             text: 'Top' + ' ' + this.chartHeader + ' ' + 'over Time',
+           },
+
+
+           xAxis: {
+             title: {
+               enabled: true,
+               text: 'Entry Date',
+             },
+
+             labels: {
+               formatter: function () {
+                 return moment.unix(this.value).format('YYYY');
+               },
+             },
+             startOnTick: true,
+             endOnTick: true,
+             tickInterval: 40 * 3600 * 1000,
+           },
+           yAxis: [{
+             title: {
+               enabled: true,
+               text: this.chartHeader,
+             },
+             labels: {
+               formatter: function () {
+                 return names[this.value];
+               }
+             },
+             tickInterval: 1,
+             lineWidth: 1,
+           },
```

Comment:
I notice the text was cut off for many of the labels on the y axis. Are we able to put a tooltip here also?

Reply:
Not sure in Highchart it has any flexibility ti add tooltip in Yaxis. I will check it out.